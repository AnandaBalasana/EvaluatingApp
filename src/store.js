import { createStore, applyMiddleware, compose } from 'redux'
import reducer from './redux/reducers'
import thunkMiddleware from 'redux-thunk'

const store = createStore(
  reducer,
  compose(applyMiddleware(thunkMiddleware),
    window.__REDUX_DEVTOOLS_EXTENSION__
      ? window.__REDUX_DEVTOOLS_EXTENSION__()
      : f => f
  )
)

export default store
