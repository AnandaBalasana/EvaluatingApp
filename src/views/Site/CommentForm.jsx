import React from 'react'
import PropTypes from 'prop-types'
import { reduxForm, Field } from 'redux-form'

class CommentForm extends React.Component {
    render() {
        return (
            <form onSubmit={this.props.handleSubmit} className="comment-form">
                <div className="form-group row">
                    <div className="col-md-8">
                        <Field
                            label=""
                            name="content"
                            component="textarea"
                            type="text"
                            placeholder="Type here..."
                            className="form-control"
                        />
                    </div>
                    <div className="col-md-4">
                        <button type="submit" className="btn btn-primary">
                            Add Comment
                        </button>
                    </div>
                </div>
            </form>
        )
    }
}

CommentForm.propTypes = {
    handleSubmit: PropTypes.func,
}

export default reduxForm({
    form: 'comment',
    // fields,
    // validate
    // initialValues: this.props.initialValues
})(CommentForm)
