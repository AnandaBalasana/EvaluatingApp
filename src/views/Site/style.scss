
.comment {
  .comment__info {
    color: #888;
    display: flex;
    justify-content: space-between
  }
  .comment__timestamp {
    font-size: .8em;
    font-style: italic;
  }
  .comment__added_by {
    font-weight: bold;
  }

  margin-bottom: 1.5em;
}
