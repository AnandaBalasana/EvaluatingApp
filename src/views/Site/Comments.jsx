import CommentForm from './CommentForm'
import PropTypes from 'prop-types'
import React from 'react'
import withToggle from '../../components/withToggle/withToggle'
import { addComment } from '../../redux/actions'
import { connect } from 'react-redux'
import { selectLoggedUser } from '../../redux/reducers/users.reducer'
import { selectComments } from '../../redux/reducers/comments.reducer'

class Comments extends React.Component {
  constructor(props) {
    super(props)
    this.formSubmit = this.formSubmit.bind(this)
  }

  formSubmit(comment) {
    comment.timestamp = new Date().toISOString().substring(0, 10)
    this.props.dispatch(addComment(comment))
  }

  renderComment(comment) {
    return (
      <div className="comment" key={comment.id}>
        <div className="comment__info">
          <div className="comment__added_by">{comment.added_by}</div>
          <div className="comment__timestamp">
            {comment.timestamp}
          </div>
        </div>
        <div className="comment__content">{comment.content}</div>
      </div>
    )
  }

  render() {
    const { comments, user } = this.props
    return (
      comments && (
        <div className="comments-container">
          <h2>Comments</h2>
          <CommentForm onSubmit={this.formSubmit} user={user} id={comments.length}/>
          {comments
            .map(comment => {
              return this.renderComment(comment)
            })}
          <button
            onClick={this.props.toggle}
            className="btn btn-primary"
          >
            {this.props.toggleStatus ? 'Hide' : 'Show all'}
          </button>
        </div>
      )
    )
  }
}

Comments.propTypes = {
  comments: PropTypes.array.isRequired,
  user: PropTypes.object,
  dispatch: PropTypes.func,
  toggle: PropTypes.func,
  toggleStatus: PropTypes.bool,
}

Comments.defauProps = {
  comments: [],
  user: {},
}

const mapStateToProps = (state, ownProps) => {
  return {
    comments: selectComments(state, ownProps.toggleStatus),
    user: selectLoggedUser(state),
  }
}


export default withToggle(connect(mapStateToProps)(Comments))
