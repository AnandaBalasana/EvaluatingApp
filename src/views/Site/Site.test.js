import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../store'
import Site from './Site'

describe('Site Page', () => {
    test('renders site component', () => {
        const { getByText } = render(
            <Provider store={store}>
                <Site />)
            </Provider>
        )
        const textElement = getByText(/Article title/i)
        expect(textElement).toBeInTheDocument()
    })

    test('snapshot', () => {
        const comp = render(
            <Provider store={store}>
                f
                <Site />
            </Provider>
        )
        expect(comp).toMatchSnapshot()
    })
})
