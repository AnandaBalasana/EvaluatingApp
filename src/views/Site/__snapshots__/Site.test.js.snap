// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Page snapshot 1`] = `
<div>
  <div
    class="site-page"
  >
    <div
      class="content"
    >
      <h1>
        Article title
      </h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus efficitur eleifend est. Quisque elementum sagittis nibh, faucibus tempor purus. Vestibulum congue lacinia massa, at hendrerit dolor rutrum vitae. Nulla in eros ut tellus semper consequat. Aenean ultricies neque ac quam vehicula, quis dictum libero auctor. Ut a nulla quis libero tincidunt mattis. Nam eu est erat. Nunc ac tempus elit. Pellentesque a ullamcorper nisi. Sed faucibus tortor feugiat orci aliquam sagittis. Nam viverra leo in massa euismod, aliquam dictum ligula tincidunt. Suspendisse at iaculis tellus, vitae maximus tortor.
      </p>
      <p>
        Vestibulum eu tellus mauris. Nullam blandit ipsum at ligula ultrices rhoncus. Donec placerat sem vehicula quam tempus rutrum. Fusce sem nulla, malesuada vitae tempus non, ullamcorper egestas nisl. Phasellus scelerisque congue egestas. Donec gravida enim sit amet tortor facilisis tristique. Proin quis eleifend leo. Nulla ex nisl, finibus quis feugiat sit amet, tincidunt et nulla. Morbi semper nibh eu metus pellentesque, in pellentesque ante aliquam. Pellentesque vitae ligula nunc. In pretium elit sapien, quis venenatis nisi cursus at. Fusce rhoncus, sem id ullamcorper pharetra, nibh tellus ultrices justo, quis hendrerit sem magna at magna.
      </p>
      <p>
        Integer ultrices quis leo vel tristique. In ornare tortor ipsum, a tristique turpis euismod at. Aliquam augue erat, consectetur laoreet lorem ut, blandit ullamcorper arcu. Fusce pulvinar, erat vitae aliquam convallis, ligula mauris consectetur ex, eget rutrum lacus lacus eu turpis. Integer in enim massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer vehicula condimentum orci, lacinia dapibus massa rutrum sed. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Fusce neque eros, fringilla tempus sem sed, eleifend dapibus lorem.
      </p>
    </div>
    <div
      class="comments-container"
    >
      <h2>
        Comments
      </h2>
      <form
        class="comment-form"
      >
        <input
          name="added_by"
          type="hidden"
          value=""
        />
        <input
          name="id"
          type="hidden"
          value=""
        />
        <input
          name="timestamp"
          type="hidden"
          value=""
        />
        <div
          class="form-group row"
        >
          <div
            class="col-md-8"
          >
            <textarea
              class="form-control"
              label=""
              name="content"
              placeholder="Type here..."
              type="text"
            />
          </div>
          <div
            class="col-md-4"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Add Comment
            </button>
          </div>
        </div>
        <div>
          <p
            class="comment-advice"
          />
        </div>
      </form>
      <div
        class="comment"
      >
        <div
          class="comment__info"
        >
          <div
            class="comment__added_by"
          >
            Ernest Hemingway
          </div>
          <div
            class="comment__timestamp"
          >
            2020-01-08
          </div>
        </div>
        <div
          class="comment__content"
        >
          Hallo! :) 
        </div>
      </div>
      <button
        class="btn btn-primary"
      >
        Show all
      </button>
    </div>
  </div>
</div>
`;
